pkg/sync/grpc/grpc_sync.go

```diff
@@ -0,0 +1,126 @@
+package grpc
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+
+	// todo - from schema push
+	"buf.build/gen/go/kavindudodan/flagd/grpc/go/sync/v1/syncv1grpc"
+	v1 "buf.build/gen/go/kavindudodan/flagd/protocolbuffers/go/sync/v1"
+
+	"github.com/open-feature/flagd/pkg/logger"
+	"github.com/open-feature/flagd/pkg/sync"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+// Prefix for GRPC URL inputs. GRPC does not define a prefix through standard. This prefix helps to differentiate
+// remote URLs for REST APIs (i.e - HTTP) from GRPC endpoints.
+const Prefix = "grpc://"
+
+type Sync struct {
+	Target string
+	Key    string
+	Logger *logger.Logger
+}
+
+func (g *Sync) Sync(ctx context.Context, dataSync chan<- sync.DataSync) error {
+	dial, err := grpc.Dial(g.Target, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		g.Logger.Error(fmt.Sprintf("Error establishing connection: %s", err.Error()))
+		return err
+	}
+
+	return g.streamListener(ctx, dial, dataSync)
+}
+
+// streamListener performs the grpc listening on provided client connection and push updates through dataSync channel
+func (g *Sync) streamListener(ctx context.Context, dial *grpc.ClientConn, dataSync chan<- sync.DataSync) error {
```

Comment:
I think we will need stream-reconnect logic.

Reply:
Yes totally agree. This is why we have a simple logic at isync contract method [1]. I kept it simple, so we can build the retry logic there. 

However, I am not confident to include try logic and related complexity with this initial version. Because that require unit tests, and may be some more configurations (making this review extensive). Should I create a follow up issue ? Or do you think we need this withthis PR :) ? 


[1] - https://github.com/open-feature/flagd/pull/297/files#diff-5bd4f361280cd88498f36d821f08f603ba13af72ad0c1ce0313d7b3fa1ed5af5R30-R37

Reply:
I will attempt to introduce the first version of re-try logic :)

Reply:
Added connection retry handler